src/engine/threaded_engine.h

```diff
@@ -432,6 +440,9 @@ class ThreadedEngine : public Engine {
   };
   /*! thread local store for bulk */
   typedef dmlc::ThreadLocalStore<BulkStatus> BulkStatusStore;
+  /*! shared_ptr to exception_ptr, used for exception handling */
+  typedef std::shared_ptr<std::exception_ptr> ExceptionRef;
```

Comment:
can we also use ExceptionRef in other locations where we use std::shared_ptr<std::exception_ptr>？

Reply:
sure